src/Kopernicus/Components/HazardousBody.cs

```diff
@@ -117,13 +118,17 @@ private void Update()
                 heat *= heatMap.GetPixelFloat((longitude + 180) * FULL_CIRCLE, (latitude + 90) * HALF_CIRCLE);
             }
 
+            // How many rounds of heating need to be applied before the timer is positive again (heating "debt" is paid
+            // off).
+            UInt32 heatingRounds = (UInt32) Math.Floor(-_heatPosition / heatInterval + 1);
+
             foreach (Part part in FlightGlobals.ActiveVessel.Parts)
```

Comment:
What is the alternative?

Reply:
I will do that that myself, I have been told by people who understand unity better than I do that it would be better to use a for loop rather than a foreach loop.

Reply:
also, I think this should be applied to all loaded vessels rather than just the active vessel. unless there is a reason for only wanting one vessel to heat up

Reply:
Yes. Heating was restricted to only the active vessel in a previous commit. I think it should be applied to all vessels in physics range. Currently stepping out of a vessel for EVA causes rapid cooldown.